src/app/rpmostree-builtin-kargs.cxx

```diff
@@ -355,6 +358,15 @@ rpmostree_builtin_kargs (int            argc,
                                                 error))
     return FALSE;
 
+  if (opt_unchanged_exit_77)
+    {
+     if (!rpmostree_has_new_default_deployment (os_proxy, previous_deployment))
+       {
+         invocation->exit_code = RPM_OSTREE_EXIT_UNCHANGED;
+         return TRUE;
+       }
+    }
+
   g_print("Kernel arguments updated.\nRun \"systemctl reboot\" to start a reboot\n");
```

Comment:
This should only be printed in the case where there is a new deployment. Right now, it's also printed if there isn't a new deployment and we're not passing `--unchanged-exit-77`.